static/app/components/modals/importDashboardFromFileModal.tsx

```diff
@@ -0,0 +1,127 @@
+import {Fragment, useState} from 'react';
+import {browserHistory} from 'react-router';
+import {css} from '@emotion/react';
+
+import {createDashboard} from 'sentry/actionCreators/dashboards';
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {Button} from 'sentry/components/button';
+import {CodeSnippet} from 'sentry/components/codeSnippet';
+import {IconUpload} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {
+  assignDefaultLayout,
+  getInitialColumnDepths,
+} from 'sentry/views/dashboards/layoutUtils';
+import {Wrapper} from 'sentry/views/discover/table/quickContext/styles';
+
+function ImportDashboardFromFileModal({
+  Header,
+  Body,
+  Footer,
+  organization,
+  api,
+  location,
+}) {
+  const [dashboardData, setDashboardData] = useState('');
+  const [validated, setValidated] = useState(false);
+
+  function validateFile(fileToUpload) {
+    if (!(fileToUpload.type === 'application/json')) {
+      addErrorMessage('The uploaded file must be JSON');
+      setValidated(false);
+      return false;
+    }
+
+    setValidated(true);
+    return true;
+  }
+
+  const handleFileChange = e => {
+    if (e.target.files) {
+      const fileToUpload = e.target.files[0];
+      if (validateFile(fileToUpload)) {
+        const fileReader = new FileReader();
+        fileReader.readAsText(fileToUpload, 'UTF-8');
+        fileReader.onload = event => {
+          const target = event.target;
+          if (target) {
+            const parsed = JSON.parse(target.result);
+            const formatted = JSON.stringify(parsed, null, '\t');
+            setDashboardData(formatted);
+          }
+        };
+      }
+    }
+  };
+
+  const handleUploadClick = async () => {
+    const dashboard = JSON.parse(dashboardData);
+
+    try {
+      const newDashboard = await createDashboard(
+        api,
+        organization.slug,
+        {
+          ...dashboard,
+          widgets: assignDefaultLayout(dashboard.widgets, getInitialColumnDepths()),
+        },
+        true
+      );
+
+      addSuccessMessage(`${dashboard.title} dashboard template successfully added`);
+      loadDashboard(newDashboard.id);
+    } catch (error) {
+      addErrorMessage('Could not upload dashboard, JSON may be invalid');
+    }
+  };
+
```

Comment:
`loadDashboard()` is copied [from another component](https://github.com/getsentry/sentry/blob/4b60067911264601458d92bdb52e5ada779445d4/static/app/views/dashboards/manage/index.tsx#L247-L255). Should I extract the function into a module and reuse in both places? 

I chose not to do so prior to review because a) it's only used in 2 places currently, b) it's small, c) i don't know the preferences/standards for DRYness at Sentry

Reply:
Agreed, I took a look at the function and I wouldn't say it needs to be extracted at all